python/lsst/drp/tasks/metadetection_shear.py

```diff
@@ -0,0 +1,286 @@
+                    },
+                ),
+                pa.field(
+                    "patch_x",
+                    pa.uint64(),
+                    nullable=False,
+                    metadata={
+                        "doc": "Column within the tract of the patch on which this measurement was made.",
+                        "unit": "",
+                    },
+                ),
+                pa.field(
+                    "patch_y",
+                    pa.uint64(),
+                    nullable=False,
+                    metadata={
+                        "doc": "Row within the tract of the patch on which this measurement was made.",
+                        "unit": "",
+                    },
+                ),
+                pa.field(
+                    "cell_x",
+                    pa.uint64(),
+                    nullable=False,
+                    metadata={
+                        "doc": "Column within the patch of the cell on which this measurement was made.",
+                        "unit": "",
+                    },
+                ),
+                pa.field(
+                    "cell_y",
+                    pa.uint64(),
+                    nullable=False,
+                    metadata={
+                        "doc": "Row within the patch of the cell on which this measurement was made.",
+                        "unit": "",
+                    },
+                ),
+                # TODO: add more field definitions here
+            ]
+        )
+
+    def runQuantum(
+        self,
+        qc: QuantumContext,
+        inputRefs: InputQuantizedConnection,
+        outputRefs: OutputQuantizedConnection,
+    ) -> None:
+        # Docstring inherited.
+        # Read the coadds and put them in the order defined by
+        # config.required_bands (note that each MultipleCellCoadd object also
+        # knows its own band, if that's needed).
+        coadds_by_band = {
+            ref.dataId["band"]: qc.get(ref) for ref in inputRefs.input_coadds
+        }
+        outputs = self.run([coadds_by_band[b] for b in self.config.required_bands])
+        qc.put(outputs, outputRefs)
+
+    def run(self, patch_coadds: Sequence[MultipleCellCoadd]) -> Struct:
+        """Run metadetection on a patch.
+
+        Parameters
+        ----------
+        patch_coadds : `~collections.abc.Sequence` [ \
+                `~lsst.cell_coadds.MultipleCellCoadd` ]
+            Per-band, per-patch coadds, in the order specified by
+            `MetadetectionShearConfig.required_bands`.
+
+        Returns
+        -------
+        results : `lsst.pipe.base.Struct`
+            Structure with the following attributes:
+
+            - ``object_catalog`` [ `pyarrow.Table` ]: the output object
+              catalog for the patch, with schema equal to `object_schema`.
+        """
+        single_cell_tables: list[pa.Table] = []
+        for single_cell_coadds in zip(*patch_coadds, strict=True):
```

Comment:
This is a loop over bands right?  We will be making a combined coadd over bands for detection, but then process the bands simultaneously to get the shear info, fluxes etc.

Reply:
This is a loop over individual cells, which each `single_cell_coadds` object a sequence of per-band single-cell coadds.

I figured we'd coadd across bands inside `process_cell`, one cell at a time.  I was assuming there'd be no reason to coadd all cells across bands up front and then process both the multi-band coadd and the per-band coadds for each cell more later.

Reply:
Ah, I see now there is a star on patch_coadds.  So this means that patch_coadds is actuallly a sequence of sequences?  Where can I find the definition of MultiCellCoadd?

Reply:
Oops, I wrote this thinking that `MultipleCellCoadd` was iterable, but it isn't (you have to iterate over `.cells`) so this was actually wrong (just pushed a fix).

Here's the definition of `MultipleCellCoadd`: https://github.com/lsst/cell_coadds/blob/main/python/lsst/cell_coadds/_multiple_cell_coadd.py

And here's the definition of `SingleCellCoadd`: https://github.com/lsst/cell_coadds/blob/main/python/lsst/cell_coadds/_single_cell_coadd.py

Those both refer to [`ImagePlanes`](https://github.com/lsst/cell_coadds/blob/main/python/lsst/cell_coadds/_image_planes.py) and [`CommonComponents`](https://github.com/lsst/cell_coadds/blob/main/python/lsst/cell_coadds/_common_components.py) from the same package.

Reply:
Thanks.

Is there an existing way to coadd these individual band SingleCellCoadds or should we extract the masked image and do it ourselves?

How do extract the image?

Reply:
(assuming we should not rely on hidden `_outer` attribute

Reply:
whoops, I see there is a property for outer

Reply:
Go ahead and extract the outer masked image or individual outer single-plane images and do it yourself; it might make sense to move that into a method on `SingleCellCoadd` for reuse at some point, but we can look into that once you've defined the behavior you want for the combination.